pkg/discovery/worker/controller_discovery_worker.go

```diff
@@ -0,0 +1,80 @@
+package worker
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/dtofactory"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/metrics"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/repository"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+)
+
+// Converts the cluster K8s controller objects to entity DTOs
+type K8sControllerDiscoveryWorker struct {
+	cluster *repository.ClusterSummary
+}
+
+func NewK8sControllerDiscoveryWorker(cluster *repository.ClusterSummary) *K8sControllerDiscoveryWorker {
+	return &K8sControllerDiscoveryWorker{
+		cluster: cluster,
+	}
+}
+
+// Controller discovery worker collects KubeController entities discovered by different discovery workers.
+// It merges the pods belonging to the same controller but discovered by different discovery workers, and aggregates
+// allocation resources usage of the same KubeController from different workers.
+// Then it creates entity DTOs for the K8s controllers to be sent to the Turbonomic server.
+func (worker *K8sControllerDiscoveryWorker) Do(kubeControllers []*repository.KubeController) ([]*proto.EntityDTO, error) {
```

Comment:
nit: maybe document that this must be run after cluster namespaces have been populated.